Makefile

```diff
@@ -3,8 +3,8 @@
 all: test 
 
 test:
-	nosetests joblib
+	py.test joblib
 
 test-no-multiprocessing:
-	export JOBLIB_MULTIPROCESSING=0 && nosetests
+	export JOBLIB_MULTIPROCESSING=0 && py.test joblib
```

Comment:
This is the output of `nosetests joblib` when i execute it from non project root (which means it ran on installed package). But the Travis CI log on current master is not the same as this. ( [Log](https://travis-ci.org/joblib/joblib/jobs/181364511) ) - looking at this it seems like tests are happening on the codebase itself. Also, if I run the tests on installed package, the coverage will not include the test files themselves and will generate a [funny report](https://coveralls.io/builds/9142355). About the data files you're talking about, we can add a `norecuredirs` option in `setup.cfg` to specify directories to be left out of test discovery and coverage calculation.